pkg/reconciler/service/install.go

```diff
@@ -38,9 +38,6 @@ func (r *Install) Invoke(ctx context.Context, chartProvider chart.Provider, task
 	}
 
 	if task.Type == model.OperationTypeDelete {
-		if task.Component == model.CRDComponent {
```

Comment:
To save unnecessary call to the deletion-reconciler when we are reconciling, it would be good to return when the Operation-Type is "Reconile", E.g.

     if task.Type == model.OperationTypeReconcile && task.Component == model. CleanupComponent {
       return nil
    }

Reply:
In that case we need to turn around the whole if-else block, but definitely good to return early.